### PR TITLE
Feature/check got to waypoint

### PR DESCRIPTION
**Pull request by hboortz**
Still needs to be double checked outside. Works totally fine running without arming.

### CODE REVIEW

File: quadcopter_brain/src/position_subscriber.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python
```
Comment (arlolinscope) [inline quadcopter_brain/src/position_subscriber.py]: remove file

File: quadcopter_brain/src/quadcopter_brain.py [hunk 2]
```diff
@@ -33,50 +32,80 @@ def __init__(self):
         self.adjust_throttle_service = rospy.ServiceProxy(
             'adjust_throttle', Empty)
 
-    def send_waypoint(self, waypoint):
+    def go_to_waypoint(self, waypoint):
```
Comment (arlolinscope) [inline quadcopter_brain/src/quadcopter_brain.py]: Call 'send_waypoint', give docstring

File: quadcopter_brain/src/quadcopter_brain.py [hunk 3]
```diff
@@ -33,50 +32,80 @@ def __init__(self):
...
+                    print("Tried % times and giving up" % (tries)) 
                 else:
-                    print("Trying again. Tries: %d" % (tries+1))
+                    print("Retrying. Tries: %d" % (tries))
+
+    def check_reached_waypoint(self, waypoint):
+        wait_time = 0
+        rospy.Subscriber("/filtered_pos", roscopter.msg.FilteredPosition,
+                         self.position_callback)
+        while not self.has_reached_waypoint(waypoint) and wait_time < 30:
```
Comment (arlolinscope) [inline quadcopter_brain/src/quadcopter_brain.py]: pass 30 as a parameter 'waittime'

File: quadcopter_brain/src/quadcopter_brain.py [hunk 4]
```diff
@@ -33,50 +32,80 @@ def __init__(self):
...
+            time.sleep(5)
+            wait_time += 5
+            print "--> Traveling to waypoint for %d seconds" % (wait_time)
+            print "--> Current position is %d, %d" % (self.current_lat,
+                                                      self.current_long)
+        if wait_time < 50: # successfully reached
```
Comment (arlolinscope) [inline quadcopter_brain/src/quadcopter_brain.py]: Make match up with the 30

File: quadcopter_brain/src/quadcopter_brain.py [hunk 5]
```diff
@@ -33,50 +32,80 @@ def __init__(self):
...
-                    print("Tried 5 times and giving up")
+                if tries == 5:
+                    print("Tried % times and giving up" % (tries)) 
                 else:
-                    print("Trying again. Tries: %d" % (tries+1))
+                    print("Retrying. Tries: %d" % (tries))
+
+    def check_reached_waypoint(self, waypoint):
+        wait_time = 0
+        rospy.Subscriber("/filtered_pos", roscopter.msg.FilteredPosition,
```
Comment (arlolinscope) [inline quadcopter_brain/src/quadcopter_brain.py]: subscribe in the init

File: quadcopter_brain/src/quadcopter_brain.py [hunk 6]
```diff
@@ -33,50 +32,80 @@ def __init__(self):
...
+            print "--> Current position is %d, %d" % (self.current_lat,
+                                                      self.current_long)
+        if wait_time < 50: # successfully reached
+            time.sleep(5) # stay at waypoint for a few seconds
+            return "Reached waypoint"
+        else:
+            return "Failed to reach waypoint" 
+
+    def has_reached_waypoint(self, waypoint):
+        error_margin = 3  # in meters
```
Comment (arlolinscope) [inline quadcopter_brain/src/quadcopter_brain.py]: pass in error margin

File: quadcopter_brain/src/quadcopter_brain.py [hunk 7]
```diff
@@ -33,50 +32,80 @@ def __init__(self):
...
+        # self.command_service(roscopter.srv.APMCommandRequest.CMD_ARM)
+        # print('Armed')
         self.command_service(roscopter.srv.APMCommandRequest.CMD_LAUNCH)
         print('Launched')
         time.sleep(5)
         self.trigger_auto_service()
         self.adjust_throttle_service()
         for waypoint in waypoints:
-            self.send_waypoint(waypoint)
+            self.go_to_waypoint(waypoint)
```
Comment (arlolinscope) [inline quadcopter_brain/src/quadcopter_brain.py]: change to 'send_waypoint'

**Comment by hboortz** [comment]
All in feature/break_out_fly_path_methods

**Pull request closed without merging**